src/middlewared/middlewared/plugins/system.py

```diff
@@ -568,11 +573,47 @@ async def info(self):
             'license': await self.middleware.run_in_thread(self._get_license),
             'boottime': datetime.fromtimestamp(psutil.boot_time()),
             'datetime': datetime.utcnow(),
-            'timezone': (await self.middleware.call('datastore.config', 'system.settings'))['stg_timezone'],
+            'birthday': birthday_date,
+            'timezone': timezone_setting,
             'system_manufacturer': manufacturer,
             'ecc_memory': ecc_memory,
         }
 
+    async def _convert_datetime(self, response, timezone_setting=timezone.utc):
+        return datetime.fromtimestamp(response.tx_time, timezone_setting)
+
+    # Sync the clock
+    @private
+    async def sync_clock(self):
```

Comment:
Dont think you fully understood Vladimir.

ntplib is synchronous which can block an event loop, which means this has to be a method, not a coroutine.

You're also mixing other synchronous calls and async calls.

Reply:
Done.